### PR TITLE
[wip]bring back old typings + refactor flag setting code

### PR DESCRIPTION
**Pull request by cakeinpanic**

### CODE REVIEW

File: libs/cdk/src/lib/zone-configuration/zone-cfg.ts [hunk 1]
```diff
@@ -84,80 +118,62 @@ function createZoneFlagsConfigurator(): ZoneConfig {
...
+     * Unpatch all related to XHR
+     **/
+    unpatchXHR: () => {
+      config.global.disable.XHR();
+      config.events.disable.UNPATCHED_EVENTS([...xhrEvent]);
+    },
+    /**
+     * Unpatch passive events https://developers.google.com/web/updates/2016/06/passive-event-listeners
+     **/
+    usePassiveScrollEvents: () => {
```
Comment (BioPhoton) [inline libs/cdk/src/lib/zone-configuration/zone-cfg.ts:132]: The naming here is not intuitive in terms of the additional unpatch event.

File: libs/cdk/src/lib/zone-configuration/event-names.ts [hunk 2]
```diff
@@ -156,8 +167,14 @@ export const websocketEvents: (keyof WebSocketEventMap)[] = [
...
-// XHREvent (group is here to make it easier to target HXR in angular applications. It contains redundant events e.g. 'error')
-export const xhrEvents = ['XHR', 'load', 'error'];
+/**
+ * Basic XHR Events
+ * [Load](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/load_event)
+ * [Error](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/error_event)
+ * There are more events you may want to unpatch https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest
+ */
+// XHREvent (group is here to make it easier to target XHR in angular applications. It contains redundant events e.g. 'error')
+export const xhrEvents = ['load', 'error'];
```
Comment (cakeinpanic) [inline libs/cdk/src/lib/zone-configuration/event-names.ts:177]: @BioPhoton I checked, there is no `XHR` event in MDN, also when request is sent using HttpClient, only two events get subscribed on – `load` and `error`, so I removed `XHR` from here

File: libs/cdk/src/lib/zone-configuration/zone-cfg.ts (hunk 1 above)
Comment (cakeinpanic) [reply libs/cdk/src/lib/zone-configuration/zone-cfg.ts]: done!
also reorganised code and interfaces here a bit, check it please

File: libs/cdk/src/lib/zone-configuration/event-names.ts (hunk 2 above)
Comment (BioPhoton) [reply libs/cdk/src/lib/zone-configuration/event-names.ts:177]: 👍

File: libs/cdk/src/lib/zone-configuration/zone-cfg.ts (hunk 1 above)
Comment (BioPhoton) [reply libs/cdk/src/lib/zone-configuration/zone-cfg.ts]: 👍

**Pull request merged**